src/pool/srv_util.c

```diff
@@ -266,25 +266,39 @@ ds_pool_map_tgts_update(struct pool_map *map, struct pool_target_id_list *tgts,
 		} else if (opc == POOL_ADD &&
 			 target->ta_comp.co_status != PO_COMP_ST_UP &&
 			 target->ta_comp.co_status != PO_COMP_ST_UPIN) {
-			/**
-			 * XXX this is only temporarily used for recovering
-			 * the DOWNOUT target back to UP after rebuild test,
-			 * so we do not update co_ver for now, otherwise the
-			 * object layout might be changed, so the ring shuffle
-			 * is based on target version. Once this is used
-			 * for reintegrate new target, co_ver should be
-			 * updated.
-			 */
 			D_DEBUG(DF_DSMS, "change target %u/%u to UP %p\n",
 				target->ta_comp.co_rank,
 				target->ta_comp.co_index, map);
 			D_PRINT("Target (rank %u idx %u) is added.\n",
 				target->ta_comp.co_rank,
 				target->ta_comp.co_index);
+			target->ta_comp.co_status = PO_COMP_ST_UP;
+			target->ta_comp.co_fseq = ++version;
+
+			if (pool_map_node_status_match(dom, ~PO_COMP_ST_UPIN) &&
+			    !pool_map_node_status_match(dom, ~PO_COMP_ST_UP)) {
+				/*
+				 * If no ranks are UPIN but any rank is UP,
+				 * the domain is UP
+				 */
```

Comment:
Yes, I think it might make sense to remove all change to rank status in this particular function and update the rank's status somewhere else.

I remember we talked about this a while ago but I'm not sure a patch was ever created for it.